RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromL1TTracksEDProducer.cc

```diff
@@ -0,0 +1,215 @@
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/RectangularPixelTopology.h"
+#include "Geometry/CommonDetUnit/interface/GeomDetType.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+
+#include "RecoTracker/CkfPattern/interface/BaseCkfTrajectoryBuilderFactory.h"
+#include "RecoTracker/CkfPattern/interface/BaseCkfTrajectoryBuilder.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTrackerEvent.h"
+#include "RecoTracker/Record/interface/TrackerRecoGeometryRecord.h"
+#include "RecoTracker/TkTrackingRegions/interface/GlobalTrackingRegion.h"
+
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/GeomPropagators/interface/StateOnTrackerBound.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/MeasurementDet/interface/MeasurementDet.h"
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <vector>
+
+class dso_hidden SeedGeneratorFromL1TTracksEDProducer : public edm::stream::EDProducer<> {
+public:
+  SeedGeneratorFromL1TTracksEDProducer(const edm::ParameterSet& cfg);
+  ~SeedGeneratorFromL1TTracksEDProducer() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  void findSeedsOnLayer(const GeometricSearchDet& layer,
+                        const TrajectoryStateOnSurface& tsosAtIP,
+                        Propagator& propagatorAlong,
+                        const TTTrack<Ref_Phase2TrackerDigi_>& l1,
+                        edm::ESHandle<Chi2MeasurementEstimatorBase>& estimatorH,
+                        unsigned int& numSeedsMade,
+                        std::unique_ptr<std::vector<TrajectorySeed>>& out) const;
+
+  void produce(edm::Event& ev, const edm::EventSetup& es) override;
+
+private:
+  const edm::EDGetTokenT<std::vector<TTTrack<Ref_Phase2TrackerDigi_>>> theInputCollectionTag_;
+  const edm::EDGetTokenT<MeasurementTrackerEvent> theMeasurementTrackerTag_;
+
+  // Minimum eta value to activate searching in the TEC
+  const double theMinEtaForTEC_;
+
+  // Maximum eta value to activate searching in the TOB
+  const double theMaxEtaForTOB_;
+
+  const double theErrorSFHitless_;
+
+  const edm::ESGetToken<MagneticField, IdealMagneticFieldRecord> mfToken_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
```

Comment:
repeat of https://github.com/cms-sw/cmssw/pull/30342#discussion_r447849362
is this the right geometry?
IIRC, the digi geometry is not misaligned.
Shouldn't this  code use ```TrackerRecoGeometryRecord```?

Reply:
I changed this to Reco but this should not matter. The geometry is only used to do the check if this is the PhaseII geometry here: https://github.com/cms-sw/cmssw/blob/712d49d1130e399eafb433e5fbae303aea980125/RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromL1TTracksEDProducer.cc#L165-L170

So it should not matter if Digi or Reco is used because this is not used in the actual reconstruction.